src/org/omegat/languagetools/LanguageToolWrapper.java

```diff
@@ -136,52 +166,21 @@ public synchronized void onProjectChanged(PROJECT_CHANGE_TYPE eventType) {
         // handle the translation here:
         translationText = StringUtil.normalizeUnicode(translationText);
 
-        List<Mark> r = new ArrayList<Mark>();
-        List<RuleMatch> matches;
-        if (ltSource != null && bRules != null) {
-            // LT knows about source and target languages both and has bitext rules.
-
-            // sourceText represents the displayed source text: it may be null (not displayed) or have extra
-            // bidi characters for display. Since we need it for linguistic comparison here, if it's null then
-            // we pull from the SourceTextEntry, which is guaranteed not to be null.
-            // It doesn't need to be normalized because OmegaT normalizes all source text to NFC on load.
-            if (sourceText == null) {
-                sourceText = ste.getSrcText();
-            }
-            matches = Tools.checkBitext(sourceText, translationText, ltSource, ltTarget, bRules);
-        } else {
-            // LT knows about target language only
-            matches = ltTarget.check(translationText);
-        }
-
-        for (RuleMatch match : matches) {
-            Mark m = new Mark(Mark.ENTRY_PART.TRANSLATION, match.getFromPos(), match.getToPos());
-            m.toolTipText = match.getMessage();
-            m.painter = PAINTER;
-            r.add(m);
-        }
-
-        return r;
-    }
-
-    public static Optional<Language> getLTLanguage(org.omegat.util.Language lang) {
-        String omLang = lang.getLanguageCode();
-        return Languages.get().stream().filter(ltLang -> omLang.equalsIgnoreCase(ltLang.getShortName())).findFirst();
+        return bridge.getMarksForEntry(ste, sourceText, translationText);
     }
 
-    private static List<Class<?>> LT_BIRULE_BLACKLIST = Arrays.asList(DifferentLengthRule.class,
-            SameTranslationRule.class, DifferentPunctuationRule.class);
-
-    /**
-     * Retrieve bitext rules for specified languages, but remove some rules, which not required in OmegaT
-     */
-    public static List<BitextRule> getBiTextRules(Language sourceLang, Language targetLang) {
-        try {
-            return Tools.getBitextRules(sourceLang, targetLang).stream()
-                    .filter(rule -> !LT_BIRULE_BLACKLIST.contains(rule.getClass())).collect(Collectors.toList());
-        } catch (Exception ex) {
-            // bitext rules can be not defined
-            return null;
+    @Override
+    public void propertyChange(PropertyChangeEvent evt) {
+        if (!Core.getProject().isProjectLoaded()) {
+            return;
+        }
+        // This property is changed in the end of configuration dialog saving,
+        // so at this point every other related properties are already changed.
```

Comment:
It's actually an accident that this works: `PropertyChangeEvent`s aren't supposed to be sent when the new value is equal to the old value, but due to how the `Preferences` backing store is implemented, they will be sent every time in the case of non-string values.

I understand the appeal of what you did here (you only check one property; you only rebuild once, after all properties have finished changing) but I'd rather not rely on this always being the case. If you want to keep things basically the same but be safer, I'd recommend defining another preference like `LANGUAGETOOL_PREFS_CHANGED_AT` where the value is a string representing the date and time of the last change to the prefs.